### PR TITLE
Sightings endpoint

### PR DESCRIPTION
**Pull request by IlyasAli1**
https://trello.com/c/cNnTTJcm/12-endpoint-to-get-all-sightings

### CODE REVIEW

File: WhaleSpotting.UnitTests/WhaleSpotting.UnitTests.csproj [hunk 1]
```diff
@@ -5,9 +5,15 @@
         <IsPackable>false</IsPackable>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Compile Remove="Controllers\GetInfoTest.cs" />
+      <Compile Remove="Services\SightingsServiceTest.cs" />
+    </ItemGroup>
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/WhaleSpotting.UnitTests.csproj]: We don't need these so they can be removed :)

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using WhaleSpotting.Models.ApiModels;
+using WhaleSpotting.Services;
+
+namespace WhaleSpotting.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class SightingsController : ControllerBase
+    {
+        //private static readonly ILogger Logger = LogManager.GetCurrentClassLogger();
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]: Commented code.....

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        private readonly ISightingService _sightings;
+
+        public SightingsController(ISightingService sightings)
+        {
+            _sightings = sightings;
+        }
+        [HttpGet]
+        public async Task<List<SightingResponseModel>> GetInfo()
+        {
+            var sightings = await _sightings.GetSightings();
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs]: We could just do this :) 
```suggestion
            return await _sightings.GetSightings();
```

File: WhaleSpotting/Models/DbModels/SightingDbModel.cs [hunk 4]
```diff
@@ -20,5 +20,7 @@ public class SightingDbModel
         public OrcaType? OrcaType { get; set; }
         public string OrcaPod { get; set; }
         public bool Confirmed { get; set; }
+
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/DbModels/SightingDbModel.cs:23]: Could you add a comment with TODO here pls :)

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    public class SightingResponseModel
+    {
+        [JsonProperty("sighted_at")]
+        public DateTime SightedAt { get; set; }
+        [JsonProperty("species")]
+        public string Species { get; set; }
+        [JsonProperty("quantity")]
+        public int? Quantity { get; set; }
+        [JsonProperty("location")]
+        public string? Location { get; set; }
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: strings are nullable without the ?, so you do't need them

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+using WhaleSpotting.Models.DbModels;
+
+namespace WhaleSpotting.Models.ApiModels
+{
+    public class SightingResponseModel
+    {
+        [JsonProperty("sighted_at")]
+        public DateTime SightedAt { get; set; }
+        [JsonProperty("species")]
+        public string Species { get; set; }
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Attributes should have new lines before them, so can you do this throughout
```suggestion
        [JsonProperty("sighted_at")]
        public DateTime SightedAt { get; set; }
        
        [JsonProperty("species")]
        public string Species { get; set; }
```

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            SightedAt = sighting.SightedAt;
+            Species = sighting.Species.ToString();
+            Quantity = sighting.Quantity;
+            Location = sighting.Location;
+            Longitude = sighting.Longitude;
+            Latitude = sighting.Latitude;
+            Description = sighting.Description;
+            OrcaType = sighting.OrcaType.ToString();
+            //OrcaPod = sighting.OrcaPod;
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Whats the reason for this being commented?

File: WhaleSpotting/Services/Helpers/JsonParser.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using WhaleSpotting.Models.ApiModels;
+using WhaleSpotting.Models.DbModels;
+
+namespace WhaleSpotting.Services.Helpers
+{
+    public interface IJsonParser
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/Helpers/JsonParser.cs]: This file isn't used at all, pls remove

File: WhaleSpotting/Services/SightingsService.cs [hunk 9]
```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using WhaleSpotting.Models.ApiModels;
+using WhaleSpotting.Models.DbModels;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+
+namespace WhaleSpotting.Services
+{
+    public interface ISightingService
```
Comment (oskarwilliams) [inline WhaleSpotting/Services/SightingsService.cs]:
```suggestion
    public interface ISightingsService
```

File: WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+using WhaleSpotting.Models.ApiModels;
+using WhaleSpotting.Services;
+using Xunit;
+
+namespace WhaleSpotting.UnitTests.Controllers
+{
+    public class SightingsControllerTests
+    {
+        private readonly SightingsController _underTest;
+        private readonly ISightingService _sightings = A.Fake<ISightingService>();
```
Comment (oskarwilliams) [inline WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs]: Could you move this above the _underTest declaration

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs (hunk 10 above)
Comment (LenaESoftwire) [reply WhaleSpotting.UnitTests/Controllers/SightingControllerTests.cs]: fixed

File: WhaleSpotting.UnitTests/WhaleSpotting.UnitTests.csproj (hunk 1 above)
Comment (IlyasAli1) [reply WhaleSpotting.UnitTests/WhaleSpotting.UnitTests.csproj]: Fixed

File: WhaleSpotting/Controllers/SightingsController.cs (hunk 2 above)
Comment (IlyasAli1) [reply WhaleSpotting/Controllers/SightingsController.cs]: Fixed

File: WhaleSpotting/Services/SightingsService.cs (hunk 9 above)
Comment (LenaESoftwire) [reply WhaleSpotting/Services/SightingsService.cs]: fixed

File: WhaleSpotting/Services/Helpers/JsonParser.cs (hunk 8 above)
Comment (LenaESoftwire) [reply WhaleSpotting/Services/Helpers/JsonParser.cs]: fixed

File: WhaleSpotting/Controllers/SightingsController.cs (hunk 3 above)
Comment (IlyasAli1) [reply WhaleSpotting/Controllers/SightingsController.cs]: Fixed

File: WhaleSpotting/Models/DbModels/SightingDbModel.cs (hunk 4 above)
Comment (IlyasAli1) [reply WhaleSpotting/Models/DbModels/SightingDbModel.cs:23]: done

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs (hunk 5 above)
Comment (IlyasAli1) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: done

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs (hunk 7 above)
Comment (LenaESoftwire) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Yes. It's just not to forget we have it in our DBmodel and not using in here. It is not in obvious why it is there as it's not in Whale Musium response

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs (hunk 6 above)
Comment (IlyasAli1) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Done

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs (hunk 7 above)
Comment (IlyasAli1) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: We'll add a TODO

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs [hunk 11]
```diff
@@ -0,0 +1,64 @@
+using Newtonsoft.Json;
+using System;
+using WhaleSpotting.Models.DbModels;
+
+namespace WhaleSpotting.Models.ApiModels
+{
+    public class SightingResponseModel
+    {
+        [JsonProperty("sighted_at")]
+
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Random new line?

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+            Species = sighting.Species.ToString();
+            Quantity = sighting.Quantity;
+            Location = sighting.Location;
+            Longitude = sighting.Longitude;
+            Latitude = sighting.Latitude;
+            Description = sighting.Description;
+            OrcaType = sighting.OrcaType.ToString();
+            Confirmed = sighting.Confirmed;
+
+            //TODO - Add the property OrcaPod if/when it's needed
```
Comment (oskarwilliams) [inline WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: I think the propery OrcaPod is needed, why was it removed?

Comment (oskarwilliams) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: It is in the Whale museum response

File: WhaleSpotting/Controllers/SightingsController.cs [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    [Route("[controller]")]
+    public class SightingsController : ControllerBase
+    {
+        private readonly ISightingsService _sightings;
+
+        public SightingsController(ISightingsService sightings)
+        {
+            _sightings = sightings;
+        }
+        [HttpGet]
```
Comment (oskarwilliams) [inline WhaleSpotting/Controllers/SightingsController.cs:20]: Can we have a new line above here

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs (hunk 11 above)
Comment (IlyasAli1) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Fixed

File: WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs (hunk 12 above)
Comment (IlyasAli1) [reply WhaleSpotting/Models/ResponseModels/SightingResponseModel.cs]: Fixed

File: WhaleSpotting/Controllers/SightingsController.cs (hunk 13 above)
Comment (IlyasAli1) [reply WhaleSpotting/Controllers/SightingsController.cs:20]: Done

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**